### PR TITLE
Add new utilities involving ARITH_POLY_NORM

### PR DESCRIPTION
**Pull request by ajreynol**
This will be required for refactoring the string arith entailment utilities to be proof producing.

### CODE REVIEW

File: src/theory/arith/arith_poly_norm.cpp [hunk 1]
```diff
@@ -191,6 +211,75 @@ bool PolyNorm::isEqualMod(const PolyNorm& p, Rational& c) const
...
+  {
+    return Node::null();
+  }
+  for (const std::pair<const Node, Rational>& m : d_polyNorm)
+  {
+    Node coeff;
+    if (isArith)
+    {
+      coeff = nm->mkConstRealOrInt(tn, m.second);
+    }
```
Comment (abdoo8080) [inline src/theory/arith/arith_poly_norm.cpp:242]: What about BV?

File: src/theory/arith/arith_poly_norm.cpp [hunk 2]
```diff
@@ -191,6 +211,75 @@ bool PolyNorm::isEqualMod(const PolyNorm& p, Rational& c) const
...
+  if (isArith)
+  {
+    multKind = Kind::MULT;
+    one = nm->mkConstRealOrInt(tn, Rational(1));
+  }
+  else if (isBv)
+  {
+    multKind = Kind::BITVECTOR_MULT;
+    one = bv::utils::mkOne(tn.getBitVectorSize());
+  }
```
Comment (abdoo8080) [inline src/theory/arith/arith_poly_norm.cpp:231]: Could we also include `addKind` and `zero`? Maybe even go HO and add a function `f` that creates constants...

**Review by abdoo8080 — APPROVED** [review]
Still waiting for  `ARITH_POLY_NORM` helper proof rule...

File: src/theory/arith/arith_poly_norm.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/arith/arith_poly_norm.cpp]: I'd prefer not adding zero here since it is possibly unnecessary node construction.

**Pull request merged**